internal/controller/compaction/reconciler.go

```diff
@@ -38,6 +38,9 @@ import (
 const (
 	// DefaultETCDQuota is the default etcd quota.
 	DefaultETCDQuota = 8 * 1024 * 1024 * 1024 // 8Gi
+
+	// SafeToEvictKey - annotation that ignores constraints to evict a pod if set to "false".
```

Comment:
I don't get your point exactly.
PDB has nothing to do with `SafeToEvictKey`. PDB will and must be respected at all time.
This annotation is used by CA which checks the pod before evicting the pod  while doing node drain or any maintenance operation.

> And when it is set to "true", then I think it means pod can be evicted even if it violates some constraints, again like pdb, or pod affinity constraints.

If this annotation is set to true then CA can evict the pod but PDB still be respected as PDB has maxUnavailable set `1`.

Reply:
And this is done on compaction pod, and there is no PDB on compaction pod.

Reply:
reason behind this is `compaction pod` should not have this annotation `cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` as this will hinder the CA ops as CA won't be able to drain the node.

Reply:
> If this annotation is set to true then CA can evict the pod but PDB still be respected as PDB has maxUnavailable set 1

Yes but I meant when it has to evict second pod for some reason, then if we set to "true", then will it allow pod eviction by "ignoring constraints" of pdb ( or some other constraint, not necessarily pdb)? 

my only concern was the word "ignoring constraints" in the comment. Because I thought we "ignore constraints" when it is set to "true" but the comment says we ignore when it is set to "false" that's why I had a doubt.  But I get your point of view. Thanks

Reply:
done.

Reply:
> First of all, this PR is related to compaction pod created by compaction job, so there is no PDB or any constraints for compaction pod.

Yeah true, but since the comment was explanation of the annotation, I was thinking from a generic point of view. 

Thanks for improving the comment.